core/status/engine.go

```diff
@@ -132,16 +135,40 @@ func (s *status) checkHealth(ctx echo.Context) error {
 	return ctx.JSON(responseCode, result)
 }
 
-func (s *status) doCheckHealth() core.Health {
+func (s *status) doCheckHealth(ctx context.Context) core.Health {
 	results := make(map[string]core.Health, 0)
+	type checkerResult struct {
+		name    string
+		results map[string]core.Health
+	}
+	// Perform health checks concurrently to speed up the process; some modules query external resources
+	resultChan := make(chan checkerResult)
+	checkCount := 0
 	s.system.VisitEngines(func(engine core.Engine) {
 		checker, ok := engine.(core.HealthCheckable)
-		if ok {
-			for name, result := range checker.CheckHealth() {
-				results[strings.ToLower(checker.Name()+"."+name)] = result
-			}
+		if !ok {
+			// does not support health checking
+			return
 		}
+		checkCount++
+		go func(checker core.HealthCheckable) {
+			subCtx, cancel := context.WithTimeout(ctx, healthCheckTimeout)
+			defer cancel()
+			result := checker.CheckHealth(subCtx)
```

Comment:
What if the checker ignores context? Should't your outer loop include a select on return channel and timeout? eg, handle timeouts here and not in each individual checker?